src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/ReflectionJsonTypeInfoOfT.cs

```diff
@@ -0,0 +1,308 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Text.Json.Reflection;
+
+namespace System.Text.Json.Serialization.Metadata
+{
+    /// <summary>
+    /// Provides JSON serialization-related metadata about a type.
+    /// </summary>
+    internal class ReflectionJsonTypeInfo<T> : JsonTypeInfo<T>
```

Comment:
Can we do this here?
```suggestion
    internal sealed class ReflectionJsonTypeInfo<T> : JsonTypeInfo<T>
```

Reply:
it's internal but I can seal it

Reply:
pending change, will be in the next iteration